src/blocks/folder-blocks/summarize/index.tsx

```diff
@@ -0,0 +1,256 @@
+            </svg>
+            Expand sections
+          </>
+        ) : (
+          <>
+            <svg
+              className="inline-block mr-1 -ml-1"
+              viewBox="0 0 16 16"
+              width="16"
+              height="16"
+            >
+              <path d="M10.896 2H8.75V.75a.75.75 0 00-1.5 0V2H5.104a.25.25 0 00-.177.427l2.896 2.896a.25.25 0 00.354 0l2.896-2.896A.25.25 0 0010.896 2zM8.75 15.25a.75.75 0 01-1.5 0V14H5.104a.25.25 0 01-.177-.427l2.896-2.896a.25.25 0 01.354 0l2.896 2.896a.25.25 0 01-.177.427H8.75v1.25zm-6.5-6.5a.75.75 0 000-1.5h-.5a.75.75 0 000 1.5h.5zM6 8a.75.75 0 01-.75.75h-.5a.75.75 0 010-1.5h.5A.75.75 0 016 8zm2.25.75a.75.75 0 000-1.5h-.5a.75.75 0 000 1.5h.5zM12 8a.75.75 0 01-.75.75h-.5a.75.75 0 010-1.5h.5A.75.75 0 0112 8zm2.25.75a.75.75 0 000-1.5h-.5a.75.75 0 000 1.5h.5z"></path>
+            </svg>
+            Collapse sections
+          </>
+        )}
+      </button>
+      <div className="h-full d-flex flex-column bg-gray-50 overflow-auto">
+        {/* <p className={`px-6 pt-3 whitespace-pre-wrap ${fileSummary ? "" : "text-gray-400"}`}>
+        Briefly, this code will {fileSummary || "..."}
+        </p> */}
+        <pre className="divide-y divide-gray-200">
+          {!sections.length && (
+            <div className="text-center text-gray-500 italic py-10">
+              Loading...
+            </div>
+          )}
+          {sections.map(({ text, name }, i) => (
+            <Section
+              key={i}
+              text={text}
+              name={name}
+              language={language}
+              explanation={sectionExplanations[i]}
+              isCollapsed={isCollapsed}
+            />
+          ))}
+        </pre>
+      </div>
+    </div>
+  );
+}
+
+const syntaxHighlighterLanguageMap = {
+  JavaScript: "javascript",
+  TypeScript: "typescript",
+} as Record<string, string>;
+const Section = ({
+  text,
+  name,
+  explanation,
+  language,
+  isCollapsed,
+}: {
+  text: string;
+  name: string;
+  language: string;
+  explanation: string | undefined;
+  isCollapsed: boolean;
+}) => {
+  const [isCollapsedLocally, setIsCollapsedLocally] =
+    useState<boolean>(isCollapsed);
+  useEffect(() => {
+    setIsCollapsedLocally(isCollapsed);
+  }, [isCollapsed]);
+
+  const hasValidName =
+    !!name &&
+    !name.includes(" ") &&
+    name !== "()" &&
+    (explanation === undefined || !!explanation?.length);
+
+  return (
+    <div
```

Comment:
Let's add `role="button"` or actually convert this to a button, if it's meant to be interactive.